drake/systems/Simulation.h

```diff
@@ -162,7 +163,7 @@ void simulate(const System& sys, double ti, double tf,
  */
 template <typename System>
 void simulate(const System& sys, double t0, double tf,
-              const typename System::template StateVectorType<double>& x0) {
+              const typename System::template StateVector<double>& x0) {
```

Comment:
This seems correct, but I have no idea why the old code even compiled at all.  Was nobody using it?